OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs

```diff
@@ -133,6 +160,36 @@ void INotifyKilled.Killed(Actor self, AttackInfo e)
 				var cost = self.Info.TraitInfo<ValuedInfo>().Cost;
 				attackerStats.KillsCost += cost;
 				defenderStats.DeathsCost += cost;
+				if (info.AddToArmyValue)
+					defenderStats.ArmyValue -= cost;
+			}
+		}
+
+		void INotifyCreated.Created(Actor self)
+		{
+			if (self.Owner.WinState != WinState.Undefined)
+				return;
+
+			if (self.Info.HasTraitInfo<ValuedInfo>() && info.AddToArmyValue)
+			{
+				var playerStats = self.Owner.PlayerActor.Trait<PlayerStatistics>();
+				var cost = self.Info.TraitInfo<ValuedInfo>().Cost;
```

Comment:
Why don't we cache this instead of querying it each time? I suppose trying to query this in `INotifyActorDisposing` is causing your issue, so that would go away as well with caching.

Reply:
Yeah, that would indeed do it.

Reply:
Caching the `cost` works but I'm not sure caching the `playerStats` will work as expected. I'm not familiar with how it works in the language but it feels like it's going to be an orphan reference or something...

Reply:
There is no problem with caching `playerStats`. You just need to make sure to update the reference on owner change.